bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/GroupItem.java

```diff
@@ -109,22 +111,36 @@ public GroupFunction getFunction() {
      * @return all members of this and all contained {@link GroupItem}s
      */
     public Set<Item> getAllMembers() {
-        Set<Item> allMembers = new HashSet<Item>();
-        collectMembers(allMembers, members);
+        Set<Item> allMembers = getMembers((Item i) -> !(i instanceof GroupItem));

```

Comment:
Now we can also use a one liner...

WRT: http://stackoverflow.com/a/31734226/7546824